code/modules/projectiles/guns/energy/special.dm

```diff
@@ -747,14 +747,19 @@ obj/item/weapon/gun/energy/ricochet/Fire(atom/target, mob/living/user, params, r
 	update_icon()
 
 /obj/item/weapon/gun/energy/bison/update_icon()
-	if(icon == 'icons/obj/gun_experimental.dmi')	//Quick fix for xenoarch guns
-		if(power_supply.charge >= power_supply.maxcharge)
-			icon_state = "bison100"
-		else if (power_supply.charge > 0)
-			icon_state = "bison50"
-		else
-			icon_state = "bison0"
-		return
+	if(power_supply.charge >= power_supply.maxcharge)
+		icon_state = "bison100"
+	else if (power_supply.charge > 0)
+		icon_state = "bison50"
+	else
+		icon_state = "bison0"
+	return
+
+/obj/item/weapon/gun/energy/bison/alien
+	name = "alien gun"
+
+/obj/item/weapon/gun/energy/bison/alien/update_icon()
+	return
```

Comment:
not sure if this one could go too since the proc is otherwise empty. need a collab's advice on that.

Reply:
This proc is not empty if not overriden. `/obj/item/weapon/gun/energy/bison/alien/update_icon()` exists and has bison code

Reply:
no I mean just the "return" line

Reply:
The return is not needed but I think it adds clarity to the code. It makes it very explicit on the coder's intentions